torchtune/models/llama3/_special_tokens.json

```diff
@@ -0,0 +1,259 @@
+
```

Comment:
I like this format a lot b/c it mirrors how tokenizers are packaged with HF - can you explicitly call this a `_special_tokens_map.json` though?

Reply:
Hmm, but should this technically be included in our repo? It gets packaged with the tokenizer we download so maybe we should just point to that one instead??

Reply:
Thus, this file would be deleted.

Reply:
That's an interesting idea, I'm just worried that gives us less control over when certain special tokens are officially supported in our library. For example, Mistral v3 has an updated special_tokens_map but we currently only have one MistralTokenizer class that doesn't yet use the new special tokens. If we loaded in the special_tokens_map.json from HF for Mistral v3 the tokenizer will not work as expected. Then we would basically need a separate tokenzier class for each iteration of the model. We could compare it against the HF special_tokens_map.json as part of our correctness check.

Reply:
I think this is the same as our guarantees around supported models in torchtune. We make no claims to support Mistral V3 ATM so users should not expect that it works OOTB, despite likely being able to load it with a little messing around. The same principle applies for tokenizers.

Reply:
And when we make claims that we fully support a model, then that means we should fully support all the special tokens needed to run that model. Hence, the special_tokens_map file that comes with the model should be guaranteed to work with torchtune.

Reply:
In addition, let's say someone wants to add a special token for their finetuned model. If we allow people to point to a file with the special tokens map supplied, it will be trivial to do so.

Reply:
OK, I am mostly convinced. I looked at the jsons that are typically on the model pages on HF and `tokenizer.json` is the one we want. The others don't seem to have the complete list of special tokens. The only other thing is we'd have to add an additional field for the path to the special tokens json for all tokenizers. I do like the ease of seeing all the special tokens just listed there on our repo though, but maybe we can just document it somewhere for each model...

Reply:
A question from elsewhere on this PR that may be relevant to this discussion.. can we just define `special_tokens` as a property of the `Tokenizer` protocol? Then we have some formal contract about where it's exposed. (I agree that we should probably not store any actual JSON files like this in the repo)